option/option.go

```diff
@@ -70,10 +70,13 @@ func WithAPIToken(token string) ClientOption {
 	return withTokenSource{"", src, nil}
 }
 
-// WithServiceAccount returns a ClientOption that specifies the credentials file to use.
+// WithServiceAccount returns a ClientOption that specifies the credentials to use.
 func WithServiceAccount(data []byte) ClientOption {
 	ts, sa, err := signer.NewServiceAccountTokenSource(data)
-	return withTokenSource{sa.ApiEndpoint, ts, err}
+	if err != nil {
```

Comment:
I think this is ok for now. We should consider moving the parsing out and just take a the service account object directly.